### PR TITLE
make versions work

### PR DESCRIPTION
**Pull request by fsmanuel**
not that pretty but it works
#2

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -56,14 +56,28 @@ fn crates(_: &mut Request) -> IronResult<Response> {
             versions.push(version.clone());
         }
 
-        format!("{{\"id\": \"{}\", \"type\":\"crate\"}}", krate.id)
+        let krate_versions = krate.versions.iter().map(|(_, v)| {
+            let mut id = String::from("");
```
Comment (fsmanuel) [inline src/main.rs:60]: I'm sure you know how to make it a one-liner

File: src/main.rs [hunk 2]
```diff
@@ -56,14 +56,28 @@ fn crates(_: &mut Request) -> IronResult<Response> {
...
+
+        format!("{{\"id\": \"{}\", \"type\":\"crate\",\"relationships\": {{\"versions\": {{\"data\": [{}]}}}}}}", krate.id, krate_versions)
     }).collect::<Vec<String>>().join(",");
 
     json.push_str(&crates);
     json.push_str("],\"included\":[");
 
     let included = versions.iter().map(|v| {
-        format!("{{\"type\": \"version\",\"id\": \"{}\", \"crate-id\": \"{}\"}}", v.id, v.crate_id)
+        let mut id = String::from("");
```
Comment (fsmanuel) [inline src/main.rs:75]: same here

File: src/main.rs [hunk 3]
```diff
@@ -56,14 +56,28 @@ fn crates(_: &mut Request) -> IronResult<Response> {
...
+        id.push_str(&v.crate_id);
+        id.push_str(&'-'.to_string());
+        id.push_str(&v.id);
+
+        format!("{{\"type\": \"version\",\"id\": \"{}\", \"crate-id\": \"{}\", \"attributes\": {{\"name\": \"{}\"}}}}", id, v.crate_id, v.id)
```
Comment (fsmanuel) [inline src/main.rs:80]: `crate-id` should be replace by relationships

``` json
"relationships": {
  "crate": {
    "data": {
      "id": v.crate_id,
      "type": "crate"
    }
  }
}
```

**Comment by steveklabnik** [comment]
Thanks so much!

**Pull request merged**